packages/react-dom-bindings/src/client/ReactFiberConfigDOM.js

```diff
@@ -1187,19 +1187,28 @@ export function startViewTransition(
     rootContainer.nodeType === DOCUMENT_NODE
       ? rootContainer
       : rootContainer.ownerDocument;
-  // $FlowFixMe[prop-missing]
-  if (typeof ownerDocument.startViewTransition !== 'function') {
+  try {
+    // $FlowFixMe[prop-missing]
+    const transition = ownerDocument.startViewTransition({
+      update() {
+        mutationCallback();
+        // TODO: Wait for fonts.
+        afterMutationCallback();
+      },
+      types: null, // TODO: Provide types.
+    });
+    transition.ready.then(layoutCallback, layoutCallback);
+    transition.finished.then(passiveCallback);
+    return true;
+  } catch (x) {
+    // We use the error as feature detection.
+    // The only thing that should throw is if startViewTransition is missing
+    // or if it doesn't accept the object form. Other errors are async.
+    // I.e. it's before the View Transitions v2 spec. We only support View
+    // Transitions v2 otherwise we fallback to not animating to ensure that
+    // we're not animating with the wrong animation mapped.
```

Comment:
Is it worth caching this? Like after the first time we know it's not available?

Reply:
Nah. It'll soon be no browser that hits this and it's not worth optimizing for older ones.